flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -256,19 +259,27 @@
     private final StreamTaskAsyncExceptionHandler asyncExceptionHandler;
 
     /**
-     * Flag to mark the task "in operation", in which case check needs to be initialized to true, so
-     * that early cancel() before invoke() behaves correctly.
+     * INITIALIZED: task constructor was called or on init state. RUNNING: task is in operation.
+     * FAILING: task is failing e.g., if an exception has occurred inside {@link #invoke()}.
+     * CANCELED: when this task is canceled. FINISHED: task successfully terminated.
```

Comment:
How about drawing a state transition diagram? Similar to ExecutionState's annotation[1], it will be clearer.

[1] https://github.com/apache/flink/blob/2e91543836d667a0b367688bb5ce290c3164479c/flink-runtime/src/main/java/org/apache/flink/runtime/execution/ExecutionState.java#L26